Mugs model

Nantiopa opened the pull request:
First draft of mugs to be develop further

dfurmans commented on Mugs_index.html:
```diff
@@ -0,0 +1,97 @@
+<html>
+  <head>
+    <title>Filling Cups with Liquids</title>
+    <style>
```
This CSS in line should be extracted to external brand new file.

Nantiopa replied:
Hello, 
It was done after our meeting, should I leave a comment each time or you already got a notification about new commit?

Nantiopa commented:
I have updated style file with some cosmetic changes, added 3rd cup and moved buttons under cups. 
Now I want to figure out:
-  how to place cups back next to each other instead of one under another (they have rearranged after removing buttons and liquid selection under the cups divs)   
- how to make liquid fill the cup from bottom to top (now when I try to achieve that the liquid visual repersentation sticks out of the cup border)

dfurmans commented on Mugs_index.html:
```diff
@@ -0,0 +1,91 @@
+    <div><button class="button" onclick="fillCup(3)">Fill Cup</button></div>
+    <div><button class="button" onclick="emptyCup(3)">Empty Cup</button></div>
+
+    <script>
```
Wer have to extract this JS code into external file and include it. 
This is a technical change which should be pretty similar to what we did for CSS.

dfurmans commented on Mugs_index.html:
```diff
@@ -0,0 +1,91 @@
+    <div><button class="button" onclick="emptyCup(3)">Empty Cup</button></div>
+
+    <script>
+      let liquidHeight1 = 0;
```
Instead of this procedural style data - we should use our Cup model. 
Please encapsulate this data and provide a constructor to create such an object - for now please keep it simple as much as you can

dfurmans commented on Mugs_index.html:
```diff
@@ -0,0 +1,91 @@
+      let liquidHeight2 = 0;
+      let maxCapacity = 90; // 90% of the cup's height
+
+      function fillCup(cupNumber) {
```
This procedural style function should be encapsulated as a method/function into Mug model. Please try to do it so after extracting JS code into external file

dfurmans commented on Mugs_index.html:
```diff
@@ -0,0 +1,91 @@
+        }
+      }
+
+      function emptyCup(cupNumber) {
```
The same here.

dfurmans reviewed (changes requested):
Thanks for you work! good flow so far. Keep going

dfurmans commented:
> I have updated style file with some cosmetic changes, added 3rd cup and moved buttons under cups. Now I want to figure out:
> 
>     * how to place cups back next to each other instead of one under another (they have rearranged after removing buttons and liquid selection under the cups divs)
> 
>     * how to make liquid fill the cup from bottom to top (now when I try to achieve that the liquid visual repersentation sticks out of the cup border)

Those questions are pure CSS base issues. 
Right now do not put a lot of attention to it. We will see how to rewrite soon. 
Please check further MR comments!

Thanks!
